src/movement/control_system.cpp

```diff
@@ -232,29 +232,56 @@ namespace robosub
                 case robosub::control::STATE_RELATIVE:
                     /*
                      * Set the current state to absolute, but update the goal
-                     * by adding in the new goal to the current goal to make it
-                     * relative to the current state.
+                     * by adding the new goal relative to the current state.
                      */
-                    this->goal_types[i] = robosub::control::STATE_ABSOLUTE;
-                    if(i < 3)
-                    {
-                        this->goals[i] = state_vector[i] + control_values[i];
-                    }
-                    else if (i == 3)
-                    {
-                        this->goals[i] = wraparound(state_vector[i] +
-                                control_values[i], -180.0, 180.0);
-                    }
-                    else if (i == 4)
+
+                    //if we just want to maintain our current state
+                    if(control_values[i] == 0)
                     {
-                        this->goals[i] = wraparound(state_vector[i] +
-                                control_values[i], -90.0, 90.0);
+                        //if the previous goal was absolute, we can just reuse
+                        //the last absolute goal
+                        if (this->goal_types[i] ==
+                            robosub::control::STATE_ABSOLUTE)
+                        {
```

Comment:
Maybe this if else should be switched with no else statement, just a comment about what it would be.